### PR TITLE
Add note about parsing images with digest

### PR DESCRIPTION
**Pull request by AbdelrahmanElawady**
Fixes: #11

### CODE REVIEW

**Comment by milosgajdos** [comment]
PTAL @thaJeztah

File: reference.go [hunk 1]
```diff
@@ -1,6 +1,9 @@
 // Package reference provides a general type to represent any way of referencing images within the registry.
 // Its main purpose is to abstract tags and digests (content-addressable hash).
 //
+// Note: Hash implementation packages must be imported in order
+// for digest parsing to work. Example package for sha256: [crypto/sha256]
+//
```
Comment (thaJeztah) [inline reference.go:6]: Perhaps we should make this a separate section below the Grammar; we could borrow some of the wording from https://pkg.go.dev/github.com/opencontainers/go-digest#readme-usage

Might also be good to include this information in GoDoc for individual functions to describe their behavior.

**Review by thaJeztah — COMMENTED** [review]
Thanks! Left a note / suggestion; I also gave it a quick go, and pushed some to https://github.com/distribution/reference/pull/13 (feel free to take (parts of) those changes if you think they're useful).

**Comment by AbdelrahmanElawady** [comment]
Well, I didn't know how verbose you wanted the note so I kept it concise. But I see #13 already described it better. So, I believe this PR is obsolete now.
I left a [suggestion](https://github.com/distribution/reference/pull/13#discussion_r1595591802) there and I will close this PR now. Thanks! :)

**Pull request closed without merging**